### PR TITLE
Fixed: torrent hashes, hard filtering on series, magnet links, announ…

### PR DESCRIPTION
**Pull request by itsvncl**
Hey there!

I found a few major issues durring the magnet link creation.
I also tried to add some fallback to hard filters, to save same poorly names torrents.

### CODE REVIEW

File: utils/get_availability.py [hunk 1]
```diff
@@ -150,21 +179,43 @@ def get_torrent_info(item, config):
         logger.error("Failed to retrieve torrent info after " + str(max_retries) + " attempts")
     logger.info("Got torrent info")
     torrent = bencode.bdecode(response.content)
-    trackers = []
+
+    #To avoid duplicates, lets use a set
+    trackers = set()
```
Comment (davidemarcoli) [inline utils/get_availability.py:184]: Seems to break json dump in `cache_torrents`

**Review by davidemarcoli — CHANGES_REQUESTED** [review]

File: utils/get_availability.py [hunk 2]
```diff
@@ -150,21 +179,43 @@ def get_torrent_info(item, config):
...
+        else:
+            trackers.add("tracker:" + torrent['announce'])
+
     files = []
     if 'files' in torrent['info']:
         for file in torrent['info']['files']:
             if file['path'][-1].lower().endswith(tuple(format)):
                 files.append(file['path'][-1])
 
-    hash = hashlib.sha1(bencode.bencode(torrent['info'])).hexdigest()
```
Comment (davidemarcoli) [inline utils/get_availability.py:166]: Hash is used on line 233 (and for rd on line 21)

**Review by davidemarcoli — CHANGES_REQUESTED** [review]

**Comment by aymene69** [comment]
Hi! Thanks for your contribution!

It seems to work fine on movies as far as I tested

The problem is with series. I searched for Simpson S10E02 and it gave me S30 results instead.

File: utils/get_availability.py (hunk 2 above)
Comment (itsvncl) [reply utils/get_availability.py:166]: Good catch! I totally forgot about the hash variable. It should be treated like the magnet link, so If any debrid like provider responds with anything other than 200 on the instantAvailability test the base32 version should be used.

**Comment by itsvncl** [comment]
> Hi! Thanks for your contribution!
> 
> It seems to work fine on movies as far as I tested
> 
> The problem is with series. I searched for Simpson S10E02 and it gave me S30 results instead.

That is 100% correct and a huge oversight from my side. I'm reverting the changes I made to searching.

What I was trying to achieve is to let torrents that are named like: "Simpsons S01-S06" appear for a search for lets say S03, since it really is in the S01-S06 time frame. 
And an even more extreme situation where the torrent name is something like: "Simpsons - COMPLETE", or maybe even just "Simpsons".

What should be done is to do the filtering at torrent level (if there is a torrent file of course).
I mean something like this:
```
response = requests.get(torrentLink)
torrent = bencode.bdecode(response.content)
filtered_files = series_file_filter(torrent['info']['files'], season, episode)
video_file = max(filtered_files, key=lambda x: x['length'])
episodeFileIdx = video_file['index']
```
Where the `series_file_filter ` is like the one I did in filter_results.py
If the chosen Season and Episode is in someway inside of the torent's files, then it should be shown.

Also in this case it might be a good idea to only show the filename **inside** of the torrent and it's size to the end user and not the entire torrent.

On foreign trackers (in my case Hungarian) this occurance of bundled series are super common.
One example is if I search for "Prison Break" on iTorrent, the only result is "Prison.Break.The.Complete.Series".

**Review by davidemarcoli — APPROVED** [review]
Now it works!

**Pull request merged**